src/OpenTelemetry/Logs/Pool/LogRecordThreadStaticPool.cs

```diff
@@ -19,15 +21,20 @@ public LogRecord Rent()
         var logRecord = Storage;
         if (logRecord != null)
         {
+            Debug.Assert(logRecord.Source == LogRecord.LogRecordSource.FromThreadStaticPool, "logRecord.Source was not FromThreadStaticPool");
             Storage = null;
             return logRecord;
         }
 
-        return new();
+        return new()
```

Comment:
nit: As it is a return for readability specify the type instead of just new.

Reply:
I rewrote it a bit to try and improve the readability. Check it out LMK